ext/RootsForwardDiffExt.jl

```diff
@@ -0,0 +1,39 @@
+module RootsForwardDiffExt
+
+using Roots
+import Roots: solve
+using ForwardDiff
+import ForwardDiff: Dual, derivative, gradient, partials
+
+
+function solve(ZP::ZeroProblem, M::Roots.AbstractUnivariateZeroMethod, p::Dual{T}; kwargs...) where {T}
+
+    f = ZP.F
+    pᵥ = p.value
```

Comment:
I'd use the ForwardDiff API for accessing the value:
```suggestion
    pᵥ = ForwardDiff.value(p)
```